src/main/java/org/damap/base/rest/invenio_damap/InvenioDamapResourceMapper.java

```diff
@@ -0,0 +1,169 @@
+      dmpDO.setDataKind(EDataKind.SPECIFY);
+      dmpDO.setReusedDataKind(EDataKind.SPECIFY);
+    }
+
+    if (madmpDataset.getTitle() != null) {
+      datasetDO.setTitle(madmpDataset.getTitle());
+    }
+
+    if (madmpDataset.getType() != null) {
+      // Setting data type
+      EDataType type = EDataType.getByValue(madmpDataset.getType());
+      if (type == null) {
+        log.info("Could not infer EDataType from provided value: " + madmpDataset.getType());
+      } else {
+        var types = Objects.requireNonNullElse(datasetDO.getType(), new ArrayList<EDataType>());
+        types.add(type);
+        datasetDO.setType(types);
+      }
+    }
+
+    return datasetDO;
+  }
+
+  private static void mapDistribution(Dataset madmpDataset, DatasetDO datasetDO, DmpDO dmpDO) {
+    if (madmpDataset.getDistribution() == null) {
+      return;
+    }
+
+    Set<String> licenses = new HashSet<>();
+    for (Distribution d : madmpDataset.getDistribution()) {
+      if (d.getDataAccess() != null) {
+        var dataAccess = EDataAccessType.getByValue(d.getDataAccess().value());
+        if (dataAccess != null && dataAccess.compare(datasetDO.getDataAccess()) == 1) {
+          datasetDO.setDataAccess(dataAccess);
+        }
+      }
+      licenses.addAll(d.getLicense().stream().map(l -> l.getLicenseRef().toString()).toList());
+
+      if (d.getByteSize() != null
+          && d.getByteSize() > Objects.requireNonNullElse(datasetDO.getSize(), 0L)) {
+        datasetDO.setSize(d.getByteSize().longValue());
+      }
+
+      if (d.getHost() == null) {
+        // Nothing more to do
+        return;
+      }
+
+      Host host = d.getHost();
+      String hostPath = host.getUrl() == null ? null : host.getUrl().getPath();
+
+      // Check if the provided storage is already set on the DMP.
+      var externalStorages = dmpDO.getExternalStorage();
+      ExternalStorageDO externalStorageDO =
+          externalStorages.stream()
+              .filter(s -> s.getUrl() != null && s.getUrl().equals(hostPath))
+              .findFirst()
+              .orElse(null);
+
+      if (externalStorageDO == null) {
+        externalStorageDO = new ExternalStorageDO();
+        externalStorageDO.setBackupFrequency(host.getBackupFrequency());
+        externalStorageDO.setStorageLocation(
+            host.getGeoLocation() != null ? host.getGeoLocation().toString() : null);
+        externalStorageDO.setTitle(host.getTitle());
+        externalStorageDO.setUrl(hostPath);
+        externalStorages.add(externalStorageDO);
+      }
+
+      var datasetHashes = externalStorageDO.getDatasets();
+      if (datasetHashes.contains(datasetDO.getReferenceHash())) {
+        datasetHashes.add(datasetDO.getReferenceHash());
+      }
+    }
+
+    // TODO: Support multiple licenses
+    Arrays.stream(ELicense.values())
+        .filter(eLicense -> licenses.contains(eLicense.getUrl()))
+        .findFirst()
+        .ifPresent(datasetDO::setLicense);
```

Comment:
For now, we should set the first license we can resolve from a distribution and ignore further licenses. The maDMP datamodel does not support multiple licenses so we should at least try to set one.

Reply:
`.findFirst()` ignores further licenses after the first match - or am i missing something?

Reply:
discussed offline. we should iterate the received licenses and match them with the enum license